### PR TITLE
Add `ReportBody` wrapper around `sgx_report_body_t`

### PR DESCRIPTION
**Pull request by nick-mobilecoin**

### CODE REVIEW

**Comment by nick-mobilecoin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #91** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/91" target="_blank">Graphite</a> 
    * **PR #86** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/86" target="_blank">Graphite</a> 
      * **PR #92** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/92" target="_blank">Graphite</a> 
        * **PR #103** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/103" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/103?utm_source=stack-comment).

File: core/types/src/report.rs
Comment (nick-mobilecoin) [inline core/types/src/report.rs:15]: Up for suggestions on these transitive types.  Don't think it's ideal they're here but still flushing out where else they might be used.

**Review by jcape — CHANGES_REQUESTED** [review]
Recommend making `Measurement` an enum and using distinct types for MRENCLAVE and MRSIGNER hash types.

File: core/types/src/report.rs
Comment (nick-mobilecoin) [reply core/types/src/report.rs]: Brought in most of the Measurement logic from mc_attest_core

File: core/types/src/report.rs
Comment (jcape) [inline core/types/src/report.rs:61]: Can this be derived?

File: core/types/src/report.rs
Comment (jcape) [inline core/types/src/report.rs:149]: Same Q: can this be derived?

File: core/types/src/macros.rs
Comment (jcape) [inline core/types/src/macros.rs:98]: Can we remove this and just derive it?

**Review by jcape — APPROVED** [review]
LGTM, but ideally we are just `derive(Default)` where applicable---the underlying FFI types should all have default on them now anyways, right?

File: core/types/src/report.rs
Comment (nick-mobilecoin) [reply core/types/src/report.rs:149]: Done as part of #115

File: core/types/src/macros.rs
Comment (nick-mobilecoin) [reply core/types/src/macros.rs:98]: Done as part of #115

**Review by awygle — APPROVED** [review]

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/103?utm_source=stack-comment).


Job Reference: C6ZyZ0Ggec8V0ksW3mYN

**Pull request merged**
